### PR TITLE
Enhance pyclass/pymodule

### PR DESCRIPTION
**Pull request by youknowone**
- Fix duplicated attribute checker for pyclass/pymodule
- `pyattr` for values from const and calcaulated (from function)
- py* macro doesn't allow mismatched item type
- pymodule inject `__module__` field of member class
- `pyattr` can explicitly create attribute for `pyclass` in `pymodule`
  - if function or method require similar work, it can be expanded.

- worked on a few modules with upper features

### CODE REVIEW

File: derive/src/pymodule.rs [hunk 1]
```diff
@@ -1,50 +1,73 @@
...
-        Meta::NameValue(_) => Err(meta),
-    }
-}
+use std::collections::HashMap;
+use syn::{parse_quote, spanned::Spanned, AttributeArgs, Ident, Item, Meta, NestedMeta};
 
 #[derive(Default)]
 struct Module {
-    items: HashSet<(ModuleItem, Vec<Meta>)>,
+    items: HashMap<(String, Vec<Meta>), ModuleItem>,
```
Comment (coolreader18) [inline derive/src/pymodule.rs:10]: Do we still have to have the `Vec<Meta>` in the key if we're distinguishing by name now?

File: derive/src/util.rs [hunk 2]
```diff
@@ -59,23 +59,41 @@ pub fn optional_attribute_arg(
...
 }
 
-pub fn module_class_name(mod_name: Option<String>, class_name: &str) -> String {
+pub(crate) fn module_class_name(mod_name: Option<String>, class_name: &str) -> String {
     if let Some(mod_name) = mod_name {
         format!("{}.{}", mod_name, class_name)
     } else {
         class_name.into()
     }
 }
```
Comment (coolreader18) [inline derive/src/util.rs:69]: We can just use the `str.strip_prefix()` method, it was stabilized in 1.45

File: derive/src/pymodule.rs (hunk 1 above)
Comment (youknowone) [reply derive/src/pymodule.rs:10]: Yes, we still need it. `Vec<Meta>` is used to allow duplicated names when it has different cfgs. We use it to allow same named function exists one with `#[unix]` and the other one with `#[windows]`

File: derive/src/util.rs (hunk 2 above)
Comment (youknowone) [reply derive/src/util.rs:69]: Sorry, I didn't get what you want to tell. It looks like optional concatnation rather than stripping.

Comment (coolreader18) [reply derive/src/util.rs:69]: Oh, whoops, meant to comment this on the `util::strip_prefix` function; the app I use isn't great about being precise like that :sweat_smile:

**Pull request merged**